components/server/ee/src/user/user-service.ts

```diff
@@ -32,10 +34,35 @@ export class UserServiceEE extends UserService {
 
         // the self-hosted case
         if (!this.licenseEvaluator.isEnabled(Feature.FeatureSetTimeout, userCount)) {
-            return "30m";
+            return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
         }
 
-        return "60m";
+        return WORKSPACE_TIMEOUT_DEFAULT_LONG;
+    }
+
+    public workspaceTimeoutToDuration(timeout: WorkspaceTimeoutDuration): string {
+        switch (timeout) {
+            case WORKSPACE_TIMEOUT_DEFAULT_SHORT:
+                return "30m";
+            case WORKSPACE_TIMEOUT_DEFAULT_LONG:
+                return this.config.workspaceDefaults.timeoutDefault || "60m";
```

Comment:
Sorry if you've already explained this elsewhere, but I wonder what the reasoning is behind:
- Allowing configuration of `WORKSPACE_TIMEOUT_DEFAULT_LONG`
- Hard-coding `WORKSPACE_TIMEOUT_DEFAULT_SHORT` to `30m`

(Also, in your Pull Request description, the example has `DEFAULT_LONG` as `30m` and `EXTENDED` as `60m`, which I found a bit confusing.)

Reply:
(see below, it's more or less what we had before with EE and non-EE)